test/node/int_32_tests.js

```diff
@@ -97,4 +98,50 @@ describe('Int32', function () {
       });
     }
   });
+
+  describe('fromString', () => {
+    const acceptedInputs = [
+      ['Int32.max', '2147483647', 2147483647],
+      ['Int32.min', '-2147483648', -2147483648],
+      ['zero', '0', 0],
+      ['non-leading zeros', '45000000', 45000000],
+      ['zero with leading zeros', '000000', 0],
+      ['positive leading zeros', '000000867', 867],
+      ['explicity positive leading zeros', '+000000867', 867],
+      ['negative leading zeros', '-00007', -7]
+    ];
+    const errorInputs = [
+      ['Int32.max + 1', '2147483648'],
+      ['Int32.min - 1', '-2147483649'],
+      ['positive integer with decimal', '2.0'],
+      ['zero with decimal', '0.0'],
+      ['negative zero', '-0'],
+      ['Infinity', 'Infinity'],
+      ['-Infinity', '-Infinity'],
+      ['NaN', 'NaN'],
+      ['fraction', '2/3'],
+      ['commas', '34,450'],
+      ['exponentiation notation', '1e1'],
+      ['octal', '0o1'],
+      ['binary', '0b1'],
+      ['hex', '0x1'],
+      ['empty string', ''],
+      ['leading and trailing whitespace', '    89   ', 89]
+    ];
+
+    for (const [testName, value, expectedInt32] of acceptedInputs) {
+      context(`when case is ${testName}`, () => {
+        it(`should return Int32 that matches expected value`, () => {
+          expect(Int32.fromString(value).value).to.equal(expectedInt32);
+        });
+      });
+    }
```

Comment:
It's generally nice for tests to describe the behavior, so it's easy to understand what we expect from test output.  Currently, the output looks like this:
```
  when case is Int32.max
    should return Int32 that matches expected value
```

there are lots of ways to write this description so it's helpful, a couple of examples:

```
// no context block, just a test description
Int32.fromString(Int32.max) returns `Int32(BSON_MAX_INT_32)`

// context block
Int32.fromString(Int32.max)
  returns an Int32 representing Int32.max

// or just simple
Int32.fromString parses successfully for
	Int32.max
    Int32.min
    ...
```

If you don't mind reworking the test titles some, I think that'd be a nice improvement, so long as the test output is descriptive

Reply:
Bookkeeping: Discussing in slack.